### PR TITLE
fix: use assembly in vcf data fetcher

### PR DESCRIPTION
**Pull request by sehilyi**
Reference: https://github.com/gosling-lang/gosling.js/issues/582#issuecomment-1131808700

Similar should be fixed in other data fetchers, too

### CODE REVIEW

File: src/data-fetcher/vcf/gosling-vcf-data.ts [hunk 1]
```diff
@@ -36,13 +38,14 @@ class GoslingVcfData {
...
-            dataConfig['chromSizesUrl'] = 'https://s3.amazonaws.com/gosling-lang.org/data/hg19.chrom.sizes';
-            dataConfig['sampleLength'] = dataConfig.sampleLength ?? 1000;
-
             return tileFunctions
-                .init(this.uid, dataConfig.vcfUrl, dataConfig.tbiUrl, dataConfig.chromSizesUrl, dataConfig.sampleLength)
+                .init(
+                    this.uid,
+                    dataConfig.url,
+                    dataConfig.indexUrl,
+                    `https://s3.amazonaws.com/gosling-lang.org/data/${dataConfig.assembly}.chrom.sizes`,
```
Comment (manzt) [inline src/data-fetcher/vcf/gosling-vcf-data.ts]: Isn't there a utility for this?

```javascript
GET_CHROM_SIZES(this.assembly).path
```

**Review by manzt — CHANGES_REQUESTED** [review]
Looks good. Just a comment to avoid hard-coding URLs. This will be important if for custom chromosomes sizes.

**Comment by manzt** [comment]
It's important to note that our dependency on these self-hosted TSVs on s3 means that Gosling can't be used offline.

File: src/data-fetcher/vcf/gosling-vcf-data.ts (hunk 1 above)
Comment (sehilyi) [reply src/data-fetcher/vcf/gosling-vcf-data.ts]: Yes, it is. Thanks for this suggestion.

**Comment by sehilyi** [comment]
Yes, I think when we support the custom assemblies, we should remove the dependencies entirely and use the inline URLs

**Review by manzt — APPROVED** [review]

**Pull request merged**